### PR TITLE
#170947438 User logout

### PR DESCRIPTION
**Pull request by blaise82**
#### What does this PR do?
- logout a user
#### Description of Task to be completed?
- create logout route
- create logout function in the user controller

#### How should this be manually tested?
- Clone the repo
- navigate to ft-user-logout-170947438 using `git checkout ft-user-logout-170947438`
- Run npm install
- DELETE `api/v1/auth/logout`
- Run `http://localhost:3000/api/v1/auth/logout` to test it.

#### Screenshots (if appropriate)
Screen Shot 2020-02-13 at 15 34 28

[finishes ]

### CODE REVIEW

File: locales/en.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
+	"loginSuccessfully": "User login successfully",
+	"signUpSuccess": "Registered successfully, an email has been sent to you, use it to verify your account",
+	"onVerifySuccess": "Your account is verified, you can now login",
+	"onVerifyFailure": "The provided email don't exist, create an account",
+	"IncorrectEmailPassword": "Incorrect username or password combination",
+	"EmailExist": "email already taken",
+	"localLoginSuccess": "Successfully logged in",
+	"internalServerError": "This service is currently unavailable. Developers at barefoot nomad are maintaining it",
+	"outSuccessfully": "outSuccessfully"
+}
```
Comment (AbdoulNuru) [inline locales/en.js]: add a new line

**Review by AbdoulNuru — CHANGES_REQUESTED** [review]

File: locales/en.js (hunk 1 above)
Comment (blaise82) [reply locales/en.js]: Done

File: src/utils/is-logged-in.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  const Token = localStorage.getItem('Token');
+  if (Token) {
+    return res.status(200).json({
+      message: req.i18n.__('Welcome')
+    });
+  }
+  return res.status(200).json({
+    message: req.i18n.__('InWelcome')
+  });
+};
```
Comment (AbdoulNuru) [inline src/utils/is-logged-in.js]: Why is it that both returns are doing the same thing?

File: src/routes/api/index.js [hunk 3]
```diff
@@ -2,13 +2,18 @@ import swaggerJsdoc from 'swagger-jsdoc';
...
 import swaggerOptions from 'utils/api-specifications';
-import welcomeRoute from './welcome';
+import isLoggedIn from '../../utils/is-logged-in';
 import userRoutes from './user.routes';
 
 const router = Router();
 const swaggerDoc = swaggerJsdoc(swaggerOptions);
 router.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerDoc));
-router.use('/', welcomeRoute);
+router.get('/', isLoggedIn);
```
Comment (AbdoulNuru) [inline src/routes/api/index.js]: Why was this route '/' removed? yet it is the default route that displays the welcome message, plus if you even check your review app it is automatically displaying the welcome message

Comment (blaise82) [reply src/routes/api/index.js]: Screen Shot 2020-02-17 at 11 58 51


@AbdoulNuru, as you can see in the image our https://octopus-bn-backend-staging.herokuapp.com is not returning any status or anything meaningful and when I was trying to create an SSL certificate, it was giving an error so to fix it I had just to implement something and return a given status. 


but you can test the main route by navigating to localhost:3000 you will still see the welcome message

File: src/routes/api/index.js [hunk 4]
```diff
@@ -2,13 +2,18 @@ import swaggerJsdoc from 'swagger-jsdoc';
 import swaggerUi from 'swagger-ui-express';
 import { Router } from 'express';
 import swaggerOptions from 'utils/api-specifications';
-import welcomeRoute from './welcome';
+import isLoggedIn from '../../utils/is-logged-in';
```
Comment (hakoemmy) [inline src/routes/api/index.js]:
```suggestion
import isLoggedIn from '../../utils/isLoggedIn';
```
Would you please consider using camelCase naming convention

File: src/utils/is-logged-in.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+
+import localStorage from 'localStorage';
+
```
Comment (hakoemmy) [inline src/utils/is-logged-in.js]: Would you please try documenting ``` isLoggedIn``` method

**Review by hakoemmy — CHANGES_REQUESTED** [review]

**Review by RusimbiPatrick — APPROVED** [review]

File: src/routes/api/index.js (hunk 3 above)
Comment (AbdoulNuru) [reply src/routes/api/index.js]: Yes that welcome route was removed by someone, but the TTL changed it back to what it was before(you can check from develop branch), so my worry is that now you are removing it again.

File: src/routes/api/index.js (hunk 4 above)
Comment (blaise82) [reply src/routes/api/index.js]: thanks

File: src/routes/api/index.js (hunk 3 above)
Comment (blaise82) [reply src/routes/api/index.js]: done  please navigate to `localhost:3000 `
before and after login

File: src/utils/is-logged-in.js (hunk 5 above)
Comment (blaise82) [reply src/utils/is-logged-in.js]: Thanks for the feedback 
isLoggedIn method now

**Review by hakoemmy — APPROVED** [review]

**Review by AbdoulNuru — APPROVED** [review]

**Review by kcharles52 — APPROVED** [review]

**Pull request merged**